docs/private-apis.md

```diff
@@ -0,0 +1,348 @@
+# Gutenberg Private APIs
```

Comment:
Might make sense to add context to the document, explaining why it exists and why we need to catalog those private APIs.